code-reviews/recipes/Go.md

```diff
@@ -2,47 +2,55 @@
 
 ## Go Style Guide
 
-```gofmt``` is the automated code format style guide for Go.
-
-[Effective Go](https://golang.org/doc/effective_go.html) describes other standard elements of Go style we follow.
+[CSE](../../CSE.md) developers follow the [Effective Go](https://golang.org/doc/effective_go.html) Style Guide.
 
 ## Linters
 
-### ```gofmt```
+### go vet
+
+`go vet` is a static analysis tool that checks for common go errors, such as incorrect use of range loop variables or misaligned printf arguments. [CSE](../../CSE.md) Go code should be able to build with no `go vet` errors.
+
+### golint
+
+[golint](https://github.com/golang/lint) can be an effetive tool for finding many issues, but it errors on the side of false positives. It is best used by developers when working on code, not as part of an automated build process.
+
+## Automatic Code Formatting
 
-All developers should run ```gofmt``` in a pre-commit hook to ensure standard formatting.
+### gofmt
 
-```gofmt``` should be run as a part of every build to enforce the common standard.
+`gofmt` is the automated code format style guide for Go.
 
-### ```go vet```
+### golangci-lint
 
-```go vet``` is a static analysis tool that checks for common go errors, such as incorrect use of range loop variables or misaligned printf arguments. [CSE](../CSE.md) Go code should be able to build with no ```go vet``` errors.
+[golangci-lint](https://github.com/golangci/golangci-lint/) is the replacement for the now depricated `gometalinter`. It is 2-7x faster than `gometalinter` [along with a host of other benefits](https://github.com/golangci/golangci-lint/#comparison).
 
+One awesome feature of `golangci-lint` is that is can be easily introduced to an existing large codebase using the `--new-from-rev COMMITID`. With this setting only newly introduced issues are flagged, allowing a team to improve new code without having to fix all historic issues in a large codebase. This provides a great path to improving code-reviews on existing solutions.
 
-### What about ```golint```?
+## Build Validation
 
-[```golint```](https://github.com/golang/lint) can be an effetive tool for finding many issues, but it errors on the side of false positives. It is best used by developers when working on code, not as part of an automated build process.
+`gofmt` should be run as a part of every build to enforce the common standard.
 
+To automate this process in Azure Devops you can add the following snippet to you `azure-pipelines.yaml` file. This will lint any scripts in the `./scripts/` folder.
 
-### Automation
+> TODO: Add example of yaml for gofmt
 
-#### `golangci-lint`
+## Pre-Commit Hooks
```

Comment:
This is basically like the previous section . I know you are not modifyin content but I don't think there should be a section for Pre-commits. All the the pre commit work should be part of the other sections. like running linters and tests etc.

Reply:
The reason I added a Pre-Commit Hooks section is because this is handled differently in different languages.  

I think the linting section should discuss preferred options for linters, the build validation section discusses how to enforce it as a build validation step to protect the develop branch, and finally the pre-commit hook discusses how to implement the most common optional pre-commit settings.

In python for example, this is most often done with something like the pre-commit package, while in other languages maybe you manually update the git hooks.

I am open for alternatives of course, but it felt pretty natural to leave it as a separate section to avoid cluttering the linting section.

Reply:
again , I think this is also a matter of personal preference (sorry!)  if I go to the linting section and we are suggesting to run gofmt before committing I wouldn't think to come to this section to see how to actually do it I would just search the internet. This is why I was suggesting we add the how in every section respectively. really up to you .